src/iterable_recursive_open_struct.rb

```diff
@@ -0,0 +1,26 @@
+
+module Metalware
+  class IterableRecursiveOpenStruct < RecursiveOpenStruct
+    def each(&block)
+      convert_hash_values_to_own_class.each(&block)
+    end
+
+    def each=(*args)
+      raise IterableRecursiveOpenStructPropertyError,
+        "Cannot set property 'each', reserved to use for iteration"
+    end
+
+    private
+
+    def convert_hash_values_to_own_class
```

Comment:
By inside `IterableRecursiveOpenStruct` do you mean within another method in an instance of the class? In that case we want to be able to call it, otherwise it's effectively useless. If not would you be able to write an example with the issue you mean so I can see what we might want to change?